### PR TITLE
Add an implementation plan template

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Description

This PR adds an implementation plan template, similar to the [project proposal template](https://github.com/WordPress/openverse/blob/19791f51c063d0979112f4b9f4eeace04c8cf5ff/docs/projects/templates/project-proposal.md) and based on the [implementation plan description](https://github.com/WordPress/openverse/tree/19791f51c063d0979112f4b9f4eeace04c8cf5ff/docs/projects#implementation-plans-status-in-rfc) of our project notes document.

I, personally, find it difficult to start with an empty page, so hopefully this will be a useful guide for folks writing implementation plans! I'm sure this could be structured better or that I might be missing some aspect; any feedback is welcome.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Review by obulat — COMMENTED** [review]
I was looking for one today! Thank you for creating this.
I'll try it out tomorrow

**Comment by zackkrida** [comment]
I'll also try this with my recognition proposal that I intend to publish tonight!

File: docs/projects/templates/implementation-plan.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+## Overview
+
+<!-- A brief one or two sentence overview of the implementation being described. -->
+
+## Outlined Steps
+
+<!-- Describe the implementation step necessary for completion. -->
+
+## Dependencies
```
Comment (zackkrida) [inline docs/projects/templates/implementation-plan.md:28]:
```suggestion
## Dependencies

### Infrastructure

<!-- Describe any infrastructure that will need to be provisioned or modified. In particular, identify associated potential cost changes. -->

```

Perhaps an explicit 'infra' section?

**Review by zackkrida — APPROVED** [review]
This is great! LGTM.

**Review by obulat — APPROVED** [review]
I looked through the other implementation plans in the repository.

If the overview is a brief one or two sentences description, we might want to also add a section for a longer description of the implementation plan.
One more section that could be useful is the "Expected outcomes", as in [^1].

[^1]: Implementation Plan: Filtering and designating results with sensitive textual content https://github.com/WordPress/openverse/blob/19791f51c063d0979112f4b9f4eeace04c8cf5ff/rfcs/detecting_sensitive_textual_content/20230308-implementation_plan_filtering_and_designating_results_with_sensitive_textual_content.md#expected-outcomes

**Comment by AetherUnbound** [comment]
> If the overview is a brief one or two sentences description, we might want to also add a section for a longer description of the implementation plan. One more section that could be useful is the "Expected outcomes"

I can change the description of "overview", and add an "expected outcomes" section.

**Pull request merged**